### PR TITLE
Data transformer in expect events

### PR DESCRIPTION
**Pull request by kasperski95**
- updated `expect_events` to accept "data" also as a dictionary
- made "Data Transformer" chapter in the docs generic so that it can be referenced from cheatcode sections

### CODE REVIEW

File: protostar/commands/test/cheatcodes/expect_events_cheatcode.py [hunk 1]
```diff
@@ -14,15 +18,35 @@
...
+            "data": NotRequired[
+                Union[
+                    List[int],
+                    Dict[
+                        DataTransformerFacade.ArgumentName,
+                        DataTransformerFacade.SupportedType,
+                    ],
+                ]
+            ],
+            "from_address": NotRequired[int],
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/commands/test/cheatcodes/expect_events_cheatcode.py:34]: - `Optional[T] == Union[T, None]` - a key must be specified, but the value can be `None`
- `NotRequired` - a dictionary doesn't have to have the key with the `NotRequired` type
https://peps.python.org/pep-0655/

[the remark this answers is not in the document]
Comment (Arcticae) [reply protostar/commands/test/cheatcodes/expect_events_cheatcode.py:34]: Got it 👍

File: protostar/commands/test/cheatcodes/expect_events_cheatcode.py [hunk 2]
```diff
@@ -33,11 +57,18 @@ def build(self) -> Callable:
...
 
-            expected_events = list(map(ExpectedEvent, raw_expected_events))
+            expected_events: List[ExpectedEvent] = []
+
+            for raw_expected_event in raw_expected_events:
+                expected_events.append(
+                    self._convert_raw_expected_event_to_expected_event(
+                        raw_expected_event
+                    )
+                )
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/commands/test/cheatcodes/expect_events_cheatcode.py]: `map` is a bit cleaner than list comprehension in this case

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM but please create task about updating 
```
        self.class_hash_to_contract_path_map: Dict[ClassHashType, Path] = {}
        self.contract_address_to_class_hash_map: Dict[AddressType, ClassHashType] = {}
```
to ContractAbi

**Comment by kasperski95** [comment]
https://github.com/software-mansion/protostar/issues/407

**Pull request merged**